src/test/java/com/github/scr/j8iterables/core/CloseableSpliteratorTest.java

```diff
@@ -0,0 +1,113 @@
+package com.github.scr.j8iterables.core;
+
+import com.github.scr.j8iterables.TestResource;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.Spliterator;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Created by scr on 2/24/17.
+ */
+public class CloseableSpliteratorTest {
+    TestResource testResource;
+    Spliterator<Integer> testSpliterator;
+    CloseableSpliterator<Integer, TestResource> closeableSpliterator;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        testResource = Mockito.spy(TestResource.class);
+        testSpliterator = Mockito.spy(testResource.spliterator());
+        Mockito.doReturn(null).when(testSpliterator).getComparator();
+        closeableSpliterator = new CloseableSpliterator<>(testResource, TestResource::close, testSpliterator);
+    }
+
+    @AfterMethod
+    public void tearDown() throws Exception {
+        testResource = null;
+        testSpliterator = null;
+    }
+
+    @Test
+    public void testTrySplit() throws Exception {
+        assertThat(closeableSpliterator.trySplit(), notNullValue());
+        Mockito.verify(testSpliterator).trySplit();
+    }
+
+    @Test
+    public void testHasCharacteristics() throws Exception {
+        assertThat("Not ordered", closeableSpliterator.hasCharacteristics(Spliterator.ORDERED));
+        Mockito.verify(testSpliterator).hasCharacteristics(Spliterator.ORDERED);
+    }
+
+    @Test
+    public void testCharacteristics() throws Exception {
+        assertThat(closeableSpliterator.characteristics(), not(0));
+        Mockito.verify(testSpliterator).characteristics();
+    }
+
+    @Test
+    public void testEstimateSize() throws Exception {
+        assertThat(closeableSpliterator.estimateSize(), is(3L));
```

Comment:
So, this is guaranteed only because my TestResource array size is 3.  I don't want to check against the backingSpliterator because I want to verify that calling the CloseableSpliterator invokes backingSpliterator's method.  I could, however compare this against TestResource.size().